keyform/views.py

```diff
@@ -25,11 +27,55 @@ def get_queryset(self):
 
 class RequestView(LoginRequiredMixin, UpdateView):
     model = Request
+    success_url = reverse_lazy('home')
     template_name = "keyform/request.html"
     form_class = EditForm
 
     def get_success_url(self):
-        return reverse('home')
+        return reverse_lazy('home')
+
+class ContactView(LoginRequiredMixin, TemplateView):
+    template_name = "keyform/contact.html"
+
+    def get_context_data(self):
+        context = super(ContactView, self).get_context_data()
+        context["buildings"] = Building.objects.all()
+        return context
+
+    def post(self, request, *args, **kwargs):
+
+        if request.user.has_perm('keyform.delete_contact'):
+            pk = request.POST['pk']
+            Contact.objects.filter(pk=pk).delete();
+
+            context = self.get_context_data(**kwargs)
+            return self.render_to_response(context)
```

Comment:
Since this is an AJAX post, you should not be returning HTML. Respond back with a JSON response instead.